### ISSUE TITLE
Support KeyValueCollection filtering in filter components

### ISSUE THREAD

**Issue by GlebDurygin** [Haulmont/jmix-ui#270]
Support KeyValueCollection filtering in the filter components

Related issues: Haulmont/jmix-ui#196

### Minimal reproducible example (Filter)
```
<data>
    <keyValueCollection id="salesDc">
        <loader id="salesLoader">
            <query>
                <![CDATA[select o.customer, o.customer.name, sum(o.amount) from demo_Order o group by o.customer]]>
            </query>
        </loader>
        <properties>
            <property class="com.company.demo.entity.Customer" name="customerEntity"/>
            <property datatype="string" name="customerName"/>
            <property datatype="decimal" name="sum"/>
        </properties>
    </keyValueCollection>
</data>
```
* AddConditionScreen
![image](https://user-images.githubusercontent.com/35256248/110933302-02a13200-8346-11eb-81d6-1c5ed2456fee.png)

* PropertyFilterConditionEdit
![image](https://user-images.githubusercontent.com/35256248/110933358-1482d500-8346-11eb-8a1d-8ea936a93d8e.png)

### Minimal reproducible example (PropertyFilter)
```
<data>
    <keyValueCollection id="salesDc">
        <loader id="salesLoader">
            <query>
                <![CDATA[select o.customer, o.customer.name, sum(o.amount) from demo_Order o group by o.customer]]>
            </query>
        </loader>
        <properties>
            <property class="com.company.demo.entity.Customer" name="customerEntity"/>
            <property datatype="string" name="customerName"/>
            <property datatype="decimal" name="sum"/>
        </properties>
    </keyValueCollection>
</data>
```
```
<propertyFilter property="customerEntity.lastName" operation="STARTS_WITH"
                dataLoader="salesLoader" operationEditable="true"/>
<propertyFilter property="customerName" operation="CONTAINS"
                dataLoader="salesLoader" operationEditable="true"/>
```
### Minimal reproducible example (DataManager)
```
<data>
    <keyValueCollection id="salesDc">
        <loader id="salesLoader">
            <query>
                <![CDATA[select o.customer, o.customer.name, sum(o.amount) from demo_Order o group by o.customer]]>
            </query>
        </loader>
        <properties>
            <property class="com.company.demo.entity.Customer" name="customerEntity"/>
            <property datatype="string" name="customerName"/>
            <property datatype="decimal" name="sum"/>
        </properties>
    </keyValueCollection>
</data>
```
```
@Install(to = "salesLoader", target = Target.DATA_LOADER)
protected List<KeyValueEntity> salesLoaderLoadDelegate(ValueLoadContext valueLoadContext) {
    ValueLoadContext.Query query = valueLoadContext.getQuery();
    query.setCondition(PropertyCondition.contains("customerName", customerNameFilter.getValue()));
    return dataManager.loadValues(valueLoadContext);
}
```

### Minimal reproducible example (DataManager)
```
ValueLoadContext.Query query = ValueLoadContext.createQuery(
        "select o.customer, o.customer.name, sum(o.amount) from demo_Order o group by o.customer")
        .setCondition(PropertyCondition.contains("customerName", "test"));
ValueLoadContext context = ValueLoadContext.create()
        .setQuery(query)
        .addProperty("customerEntity")
        .addProperty("customerName")
        .addProperty("sum");
dataManager.loadValues(context);
```

### QA
Test project - 
[key-value-filtering.zip](https://github.com/Haulmont/jmix-ui/files/6129609/key-value-filtering.zip)

Filter does not support key value property that contains aggregate function (COUNT, SUM, AVG, MIN, MAX).

Check that `PropertyFilter` and `Filter` components support `KeyValueCollection` filtering

**Comment by maistrenkoIulia** [comment]
Jmix version: 1.0.0-SNAPSHOT
Jmix Studio plugin version: 1.0.SNAPSHOT2681-202
IntelliJ version: IntelliJ IDEA 2020.3.2 (Community Edition)
TC:
run attached in ticket app>key value browser
add search condition> add property
![image](https://user-images.githubusercontent.com/66992566/113155834-831ec880-924a-11eb-8717-fec10b8327ce.png)
Ok
AR:
```
java.lang.NullPointerException
	at io.jmix.ui.component.impl.AbstractSingleFilterComponent.updateCaption(AbstractSingleFilterComponent.java:220)
	at io.jmix.ui.component.impl.PropertyFilterImpl.updateCaption(PropertyFilterImpl.java:280)
	at io.jmix.ui.component.impl.PropertyFilterImpl.setOperationInternal(PropertyFilterImpl.java:183)
	at io.jmix.ui.component.impl.PropertyFilterImpl.setOperation(PropertyFilterImpl.java:150)
	at io.jmix.ui.component.propertyfilter.PropertyFilterConverter.convertToComponent(PropertyFilterConverter.java:67)
	at io.jmix.ui.component.propertyfilter.PropertyFilterConverter.convertToComponent(PropertyFilterConverter.java:39)
	at io.jmix.ui.action.filter.FilterAddConditionAction.lambda$initDefaultSelectHandler$1(FilterAddConditionAction.java:158)
	at io.jmix.ui.screen.StandardLookup.lambda$select$3(StandardLookup.java:197)
	at io.jmix.ui.util.SuccessOperationResult.then(SuccessOperationResult.java:40)
	at io.jmix.ui.screen.StandardLookup.select(StandardLookup.java:197)
	at io.jmix.ui.app.filter.condition.AddConditionScreen.onEditScreenAfterCommit(AddConditionScreen.java:201)
	at io.jmix.core.common.event.EventHub.publish(EventHub.java:170)
	at io.jmix.ui.screen.Screen.fireEvent(Screen.java:124)
	at io.jmix.ui.screen.Screen.close(Screen.java:330)
	at io.jmix.ui.screen.StandardEditor.lambda$closeWithCommit$11(StandardEditor.java:590)
	at io.jmix.ui.util.SuccessOperationResult.compose(SuccessOperationResult.java:35)
	at io.jmix.ui.screen.StandardEditor.closeWithCommit(StandardEditor.java:590)
	at io.jmix.ui.screen.StandardEditor.commitAndClose(StandardEditor.java:543)
	at io.jmix.core.common.event.EventHub.publish(EventHub.java:170)
	at io.jmix.ui.action.BaseAction.actionPerform(BaseAction.java:220)
	at io.jmix.ui.component.impl.ButtonImpl.buttonClicked(ButtonImpl.java:75)
	at io.jmix.ui.widget.JmixButton.fireClick(JmixButton.java:77)
	at com.vaadin.ui.Button$1.click(Button.java:57)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.vaadin.server.ServerRpcManager.applyInvocation(ServerRpcManager.java:153)
	at com.vaadin.server.ServerRpcManager.applyInvocation(ServerRpcManager.java:115)
	at com.vaadin.server.communication.ServerRpcHandler.handleInvocation(ServerRpcHandler.java:431)
	at com.vaadin.server.communication.ServerRpcHandler.handleInvocations(ServerRpcHandler.java:396)
	at com.vaadin.server.communication.ServerRpcHandler.handleRpc(ServerRpcHandler.java:260)
	at com.vaadin.server.communication.UidlRequestHandler.synchronizedHandleRequest(UidlRequestHandler.java:82)
	at com.vaadin.server.SynchronizedRequestHandler.handleRequest(SynchronizedRequestHandler.java:40)
	at com.vaadin.server.VaadinService.handleRequest(VaadinService.java:1607)
	at com.vaadin.server.VaadinServlet.service(VaadinServlet.java:449)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:750)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:327)
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:115)
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:81)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:119)
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:113)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:126)
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:81)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:105)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:101)
	at org.springframework.security.web.authentication.rememberme.RememberMeAuthenticationFilter.doFilter(RememberMeAuthenticationFilter.java:92)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:149)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.session.ConcurrentSessionFilter.doFilter(ConcurrentSessionFilter.java:147)
	at org.springframework.security.web.session.ConcurrentSessionFilter.doFilter(ConcurrentSessionFilter.java:125)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:103)
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:89)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:90)
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:75)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:110)
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:80)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:55)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:336)
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:211)
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:183)
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:358)
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:271)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:119)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:202)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:542)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:143)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:374)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:888)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1597)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by GlebDurygin** [comment]
Will be fixed in #419

**Comment by maistrenkoIulia** [comment]
virified
